### PR TITLE
Training animation with A.I. bot

### PR DESCRIPTION
**Pull request by breville**
This updates the training scene to have an animation with the A.I. bot, inspired by the training scene in [AI for Oceans](https://code.org/oceans).

https://user-images.githubusercontent.com/2205926/112196939-c46b0480-8bc8-11eb-903f-6a12959041a1.mov

### CODE REVIEW

File: src/UIComponents/TrainModel.jsx [hunk 1]
```diff
@@ -23,20 +34,103 @@ class TrainModel extends Component {
...
+
+  updateAnimation = () => {
+    if (this.state.frame === 15) {
+      this.setState({ headOpen: true });
+    }
+
+    this.setState({ frame: this.state.frame + 1 });
+  };
+
+  componentWillUnmount = () => {
```
Comment (made-line) [inline src/UIComponents/TrainModel.jsx:54]: First time seeing componentWillUnmount out in the wild. Very nice!

File: src/redux.js [hunk 2]
```diff
@@ -1161,7 +1167,11 @@ export function getPanelButtons(state) {
       next = { panel: "results", text: "Continue" };
     }
   } else if (state.currentPanel === "results") {
-    prev = { panel: "dataDisplayLabel", text: "Back" };
+    prev = isPanelEnabled(state, "dataDisplayLabel")
+      ? { panel: "dataDisplayLabel", text: "Back" }
+      : isPanelEnabled(state, "dataDisplayFeatures")
+      ? { panel: "dataDisplayFeatures", text: "Back" }
+      : null;
```
Comment (made-line) [inline src/redux.js:1174]: Interesting use of nested ternary operators! I'm trying to understand why this was added. Does this go hand in hand with the changes in src/index.js?

**Review by made-line — APPROVED** [review]
Nice work! I like how the animation can be dynamically resized.

File: src/redux.js (hunk 2 above)
Comment (breville) [reply src/redux.js:1174]: Good catch.  This was an opportunistic fix for the `hideSelectLabel` mode; I noticed when using the tool to build this new feature that it was still possible to get back to the label scene when it should not have been possible.

**Comment by breville** [comment]
@made-line Thanks for checking resize.  I must admit to not doing anything too explicit to support resizing, but also noticed that it seemed to work pretty well for a variety of screen widths and heights, which is always important for our UI.

File: src/UIComponents/DataTable.jsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
+/* React component to handle displaying imported data. */
```
Comment (Erin007) [inline src/UIComponents/DataTable.jsx:1]: Hooray for smaller, extracted components!

**Review by Erin007 — APPROVED** [review]
Did Mark give a thumbs up too?

**Comment by breville** [comment]
We have a new background image from @markabarrett so I take that as approval.

**Pull request merged**